hs_file_types/models/base.py

```diff
@@ -773,3 +1141,181 @@ def logical_delete(self, user, delete_res_files=True):
             # this should also delete on all metadata elements that have generic relations with
             # the metadata object
             metadata.delete()
+
+    def remove_aggregation(self):
+        """Deletes the aggregation object (logical file) *self* and the associated metadata
+        object. However, it doesn't delete any resource files that are part of the aggregation."""
+
+        # delete associated metadata and map xml document
+        istorage = self.resource.get_irods_storage()
+        if istorage.exists(self.metadata_file_path):
+            istorage.delete(self.metadata_file_path)
+        if istorage.exists(self.map_file_path):
+            istorage.delete(self.map_file_path)
+
+        # first need to set the aggregation for each of the associated resource files to None
+        # so that deleting the aggregation (logical file) does not cascade to deleting of
+        # resource files associated with the aggregation
+        for res_file in self.files.all():
+            res_file.logical_file_content_object = None
+            res_file.save()
+
+        # delete logical file (aggregation) first then delete the associated metadata file object
+        # deleting the logical file object will not automatically delete the associated
+        # metadata file object
+        metadata = self.metadata if self.has_metadata else None
+        super(AbstractLogicalFile, self).delete()
+        if metadata is not None:
+            # this should also delete on all metadata elements that have generic relations with
+            # the metadata object
+            metadata.delete()
+
+    def create_aggregation_xml_documents(self, create_map_xml=True):
+        """Creates aggregation map xml and aggregation metadata xml files
+        :param  create_map_xml: if true, aggregation map xml file will be created
+        """
+
+        log = logging.getLogger()
+
+        # create a temp dir where the xml files will be temporarily saved before copying to iRODS
+        tmpdir = os.path.join(settings.TEMP_FILE_DIR, str(random.getrandbits(32)), uuid4().hex)
+        istorage = self.resource.get_irods_storage()
+        try:
+            os.makedirs(tmpdir)
+        except OSError as ex:
+            if ex.errno == errno.EEXIST:
+                shutil.rmtree(tmpdir)
+                os.makedirs(tmpdir)
+            else:
+                raise Exception(ex.message)
```

Comment:
We lose information if we raise a generic Exception by only passing in the original exceptions message.  You can raise the offending Exception with just `raise`
Having said that, I'd prefer to just check if the directory exists and delete it if it exists before making the directory.

Reply:
Will fix it.